meteor/client/ui/RundownView/RundownTiming/RundownTimingProvider.tsx

```diff
@@ -56,16 +62,39 @@ export const RundownTimingProvider = withTracker<
 >((props) => {
 	let rundowns: Array<Rundown> = []
 	let parts: Array<Part> = []
+	let segments: Array<DBSegment> = []
 	const partInstancesMap = new Map<PartId, PartInstance>()
 	let currentRundown: Rundown | undefined
+	const segmentEntryPartInstances: PartInstance[] = []
 	if (props.playlist) {
 		rundowns = RundownPlaylistCollectionUtil.getRundowns(props.playlist)
-		const { parts: incomingParts } = RundownPlaylistCollectionUtil.getSegmentsAndPartsSync(props.playlist)
+		const { parts: incomingParts, segments: incomingSegments } = RundownPlaylistCollectionUtil.getSegmentsAndPartsSync(
+			props.playlist
+		)
 		parts = incomingParts
+		segments = incomingSegments
 		const partInstances = RundownPlaylistCollectionUtil.getActivePartInstances(props.playlist)
 
 		const currentPartInstance = partInstances.find((p) => p._id === props.playlist?.currentPartInstanceId)
+		const previousPartInstance = partInstances.find((p) => p._id === props.playlist?.previousPartInstanceId)
+
 		currentRundown = currentPartInstance ? rundowns.find((r) => r._id === currentPartInstance.rundownId) : rundowns[0]
+		// These are needed to retrieve the start time of a segment for calculating the remaining budget, in case the first partInstance was removed
+		segmentEntryPartInstances.push(
+			..._.compact([
```

Comment:
This can be done without underscore, using `[].filter(Boolean)`.

Reply:
Also, this feels like it should be optimized, no reason to scan the PartInstances collection this many times, especially since on most takes the currentPartInstance will be set.

Reply:
`filter(Boolean)` doesn't narrow the type down. How about if I added this to lib/lib?
``` ts
export function compact<T>(array: (T | false | undefined | null | '' | 0)[]): T[] {
	return array.filter(Boolean) as T[]
}
```
As for the second comment, SegmentInstances would solve it. Otherwise the Ids of the entry partInstances would need to be stored somewhere else, perhaps on the playlist